crates/tauri-cli/src/build.rs

```diff
@@ -172,19 +172,36 @@ pub fn setup(
         ));
     }
 
+    // Issue #13287 - Allow the use of target dir inside frontendDist/distDir
+    // https://github.com/tauri-apps/tauri/issues/13287
+    let mut target_path = get_cargo_target_dir(&options.args)?;
+    target_path = tauri_path.join(target_path);
+    let target_path = fs::canonicalize(target_path)?;
+
     let mut out_folders = Vec::new();
-    for folder in &["node_modules", "src-tauri", "target"] {
-      if web_asset_path.join(folder).is_dir() {
+    if target_path.starts_with(web_asset_path) {
+      // Get the path of target_path relative to web_asset_path
+      if let Ok(relative_path) = target_path.strip_prefix(web_asset_path) {
+        let relative_str = relative_path.to_string_lossy();
+        if !relative_str.is_empty() {
+          out_folders.push(relative_str.to_string());
+        }
+      }
+    }
+
+    for folder in &["node_modules", "src-tauri"] {
+      let sub_path = web_asset_path.join(folder);
+      // If the subfolder exists and its canonicalized path equals the target path, add to out_folders
+      if sub_path.is_dir() && fs::canonicalize(&sub_path)? == target_path {
```

Comment:
this check is incorrect, we check if they are the node_modules or src-tauri folders, neither are the cargo target path. In other words, we just need the is_dir check here like it was before.